vowpalwabbit/parse_example.cc

```diff
@@ -76,13 +74,14 @@ class TC_parser
   uint32_t _hash_seed;
   uint64_t _parse_mask;
   std::array<std::vector<std::shared_ptr<feature_dict>>, NUM_NAMESPACES>* _namespace_dictionaries;
+  VW::io::logger* logger;
```

Comment:
I am guessing are we making this a pointer because we parse many examples and don't want to copy the logger every time?

Reply:
Yep, that is what I was thinking. And the lifetime of the object is well known so we don't need to worry it being invalidated